src/components/Button/theme.ts

```diff
@@ -0,0 +1,244 @@
+import {CSSProperties} from 'react';
+
+type StateProperties = 'default' | 'hover' | 'focus' | 'active' | 'disabled';
+type StyleProperties =
+  | 'text'
+  | 'icon'
+  | 'background'
+  | 'border'
+  | 'gradient'
+  | 'shadow';
+
+type ComponentThemeProperties = {
+  [Theme in StateProperties]?: {[Style in StyleProperties]?: string | string[]}
+};
+
```

Comment:
Ping @tim, this is feeling pretty good in terms of developer experience. Good call on strictly typing these with string literals.